lib/src/signed_video_h26x_auth.c

```diff
@@ -759,11 +758,13 @@ maybe_validate_gop(signed_video_t *self, h26x_nalu_t *nalu)
       if (item->nalu && item->nalu->is_gop_sei && item->validation_status == 'P') {
         const uint8_t *tlv_data = item->nalu->tlv_data;
         size_t tlv_size = item->nalu->tlv_size;
-        status = tlv_find_tag_and_decode(self, tlv_data, tlv_size, PUBLIC_KEY_TAG, false);
-        if (status == SVI_OK) {
+        svi_rc ret = SVI_UNKNOWN;
+        ret = tlv_find_tag_and_decode(self, tlv_data, tlv_size, PUBLIC_KEY_TAG, false);
+        if (ret == SVI_OK) {
           public_key_found = true;
+          break;
         } else {
-          DEBUG_LOG("Public key missing (error %d)", status);
+          DEBUG_LOG("Public key missing");
```

Comment:
I think this should be under the if at line 772 instead. Which then can help simplify the code above. For example
```
public_key_found = tlv_find_tag_and_decode(self, tlv_data, tlv_size, PUBLIC_KEY_TAG, false) == SVI_OK;
```
med ändring i `while()`
```
while (item && !public_key_found) {
```